client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -368,10 +373,11 @@
 (rf/reg-event-db
  :scenarios/edit-change
  in-scenarios
- (fn [db [_ change]]
+ (fn [db [_ change cancel-next-state]]
```

Comment:
The whole state management needs a full rewrite after careful thinking, but...

In this case you can omit this extra parameter as it should copy the current state, which you can read as `(:view-state db)`. You can use that to save the "previous" state that you want to restore after closing the dialog.